include/arch/arm/arch.h

```diff
@@ -200,6 +200,46 @@ extern "C" {
 #define _ARCH_THREAD_STACK_BUFFER(sym) \
 		((char *)(sym) + MPU_GUARD_ALIGN_AND_SIZE)
 
+#ifdef CONFIG_USERSPACE
+#ifdef CONFIG_ARM_MPU
+#ifndef _ASMLANGUAGE
+#include <arch/arm/cortex_m/mpu/arm_mpu.h>
+
+#define K_MEM_PARTITION_P_NA_U_NA	(NO_ACCESS | NOT_EXEC)
```

Comment:
Would it be possible to add additional defines for regions that should be execute-enabled?
I think dynamic loading of application modules is a use-case for this.
You can do this in another patch if you like, won't gate this one.

Reply:
OK, I'll create another PR to add them after this PR is merged.